File: src/main/java/com/IceCreamQAQ/Yu/loader/SpawnClassLoader.java
package com.IceCreamQAQ.Yu.loader;

import javax.inject.Inject;
import javax.inject.Named;

public class SpawnClassLoader extends ClassLoader {

    private Long time;

    @Inject
    public SpawnClassLoader(@Named("appClassLoader") ClassLoader appClassLoader) {
        super(appClassLoader);
    }

    public Class<?> define(String name, byte[] data) {
        return defineClass(name, data, 0, data.length);
    }


    @Override
    public String toString() {
        return "InvokerClassLoader " + time;
    }
}

File: enhance/classloader/src/main/kotlin/rain/classloader/SpawnClassLoader.kt
package rain.classloader

class SpawnClassLoader(classLoader: ClassLoader) : ClassLoader(classLoader), IRainClassLoader {

    override fun define(name: String, data: ByteArray): Class<*> {
        return defineClass(name, data, 0, data.size)
    }


    override fun forName(name: String, initialize: Boolean): Class<*> {
        return loadClass(name, initialize)
    }
}
